source/process/documentation-UItext-guidelines.rst

```diff
@@ -0,0 +1,193 @@
+
+Notification messages
+~~~~~~~~~~~~~~~~~~~~~~
+
+A notification message is for informing somebody about something that happened. Such messages do not need any user input, and do not prevent a user from continuing to use Mattermost.
+
+- Use either a complete sentence or a sentence phrase. 
+- If using a complete sentence, end it with a period.
+- Examples:
+
+  - *Member added to channel*
+  - *The plug-in was installed.*
+
+Confirmation messages
+~~~~~~~~~~~~~~~~~~~~~~
+
+A confirmation message is for asking somebody to confirm whether the immediately preceding command should be proceeded with. The user cannot use Mattermost until a confirmation is given or denied.
+
+- Use complete sentences.
+- Include a question that has a Yes/No answer.
+- Examples:
+
+  - *Are you sure you want to delete this channel?*
+  - *A plug-in with this ID already exists. Would you like to overwrite it?*
+
+Warning messages
+~~~~~~~~~~~~~~~~
+
+A warning message is for alerting somebody about something that might go wrong. The user can continue using Mattermost unless the warning message needs an explicit user input.
+
+- Use complete sentences.
+- Explain what has happened or can happen, and what can go wrong as a consequence.
+- If the message contains a question, phrase it in such a manner so that it has a Yes/No answer (unless you have specific action buttons for the message).
+
+- Examples:
+
+  - *The Enterprise license expires in 2 days. If you do not renew it, some features will be disabled on licence expiry.*
+  - *If you claim this AD/LDAP account, you will no longer be able to log in with your email. Do you want to continue?*
+
+Error messages
+~~~~~~~~~~~~~~~
+
+An error message is for telling somebody that something went wrong. Errors prevent a user from doing a task or accessing a feature till the error is resolved.
+
+- Use complete sentences.
+- If what went wrong isn't obvious, explain in one sentence.
+- If a solution or workaround isn't obvious, suggest one.
+- Examples:
+
+  - *This message is too long. Shorten it to 120 characters.*
+  - *The passwords do not match.*
+
+
+Guidelines for UI elements
+---------------------------
+Use this table when writing the text for UI elements such as windows, dialog boxes, labels, and prompts.
+
+.. list-table::
+   :widths: 10 30 10 15 35
+   :header-rows: 1
+
+   * - Element
+     - Image 
+     - Capitalization
+     - Phrasing
+     - Examples     
+   * - Menu
+     - |menu.png|
+     - All words
+     - - Noun, noun phrase, or verb
+       - No punctuation
+       - Not more than three words
+     - - *Members*
+       - *Account Preferences*
+       - *Log Out*
+   * - Tooltip
+     - |tooltip.png|
+     - First word
+     - - Sentence fragment or sentence
+       - No punctuation
+       - Include articles (*a*, *an*, *the*)
+     - - *Start a Zoom meeting*
+       - *Flag for following up*
+       - *Remove from this list*
+   * - In-field text
```

Comment:
What should be included here? For example, should in-field text be an example of an entry, a description of the field, the title of the field etc? ie for an email box it could be

- johndoe@example.com
- "Email"
- "Enter an email" 
- etc

Reply:
Thing is, good accessibility practices say that we shouldn't have in-field text; rather, the text should be outside of the text box. So, I am a bit ambivalent towards having too much guidance around in-field text. But if we need that text, I would prefer to have an example of an entry (johndoe@example.com). The title of the field should, ideally, have been a label, and the dsecription of the field should, ideally, be help text.

Reply:
Agree with the example proposal, let's add that as the preference if we're in agreement

Reply:
Updated to "Sentence fragment, sentence, or word that's an example of a valid entry".
Also updated the Label guidance to say "Noun, verb, or sentence fragment that's the title of the UI element"